### PR TITLE
Implement the ability to show routes based on style provided in `RoutesPresentationStyle`.

### PR DESCRIPTION
**Pull request by MaximAlien**
Closing #3548.

### CODE REVIEW

File: Sources/MapboxNavigation/NavigationMapView.swift
Comment (MaximAlien) [inline Sources/MapboxNavigation/NavigationMapView.swift:172]: I took an opportunity to reformat comments in `NavigationMapView` to be a bit shorter. It's pretty inconvenient to use horizontal scrolling while reading them.

File: Sources/MapboxNavigation/NavigationMapView.swift
Comment (1ec5) [inline Sources/MapboxNavigation/NavigationMapView.swift:131]: Does this need to be a preference? It seems pretty reasonable to fit the camera to all the provided routes, unless we’re planning to use this method when displaying alternative routes during turn-by-turn navigation. If a caller doesn’t want to show any alternative routes, it wouldn’t pass in any alternative routes.

File: CHANGELOG.md
Comment (1ec5) [inline CHANGELOG.md:27]:
```suggestion
* Renamed the `NavigationMapView.showcase(_:animated:)` method to `NavigationMapView.showcase(_:routesPresentationStyle:animated:)`, with an optional parameter to control whether the camera fits to unselected routes in addition to the selected route. ([#3556](https://github.com/mapbox/mapbox-navigation-ios/pull/3556))
```

File: Sources/MapboxNavigation/NavigationMapView.swift
Comment (S2Ler) [reply Sources/MapboxNavigation/NavigationMapView.swift:172]: I'm with @MaximAlien here. It much easier to read documentation that is wrapped in a reasonable column width (80-120 is most commonly used). Besides this there are other advantages:
- Easier in read on GitHub for Web.
- Works without wrapping setting in the editor.
- Human can wrap text better. 
- Anything else? I've seen many other repos doing the same thing (including the ones from Apple).

**Comment by S2Ler** [comment]
> # Breaking Changes in MapboxNavigation
> # Breaking API Changes
> ## `NavigationMapView`
> * _removed_ method: `showcase(_:animated:)` in `NavigationMapView`

At this point we will have to bump base version. Otherwise this will appear in every PR after this lands. 

It is possible to not introduce such breaking change though. Adding a new method instead of modifying the old one. Even adding a new default parameter IS a breaking change for some code. An example:

```swift
import UIKit

// MARK: - v1 API

//func breakingMethod(param1: Bool) {
//
//}

// MARK: - v2 API

func breakingMethod(param1: Bool, param2: Bool = false) {

}

breakingMethod(param1: true)
let methodReference = breakingMethod(param1:) `#1`
```

`#1` line works in v1 version of the API but doesn't work in v2.

File: Sources/MapboxNavigation/RouteDurationAnnotationTailPosition.swift
Comment (MaximAlien) [reply Sources/MapboxNavigation/RouteDurationAnnotationTailPosition.swift:6]: Agree, renamed.

File: Sources/MapboxNavigation/NavigationMapView.swift
Comment (MaximAlien) [reply Sources/MapboxNavigation/NavigationMapView.swift:1213]: Good catch. I've walked through all existing occurrences and changed them to use markdown style. Seems to be working with Xcode `13.1`.

File: Sources/MapboxNavigation/NavigationMapView.swift
Comment (MaximAlien) [reply Sources/MapboxNavigation/NavigationMapView.swift:131]: Since this method might be already used by the users, to play it safe I'd rather keep using presentation of only single route by default. On CarPlay, on the other hand, we'll be presenting all routes, and fit camera to all of them. 

Unfortunately this is a drawback of extending APIs in minor releases: we can try to keep this method but introduce another one. At the same time we might end up with multiple methods, which essentially do the same work, but with only slight difference in signatures.

**Comment by MaximAlien** [comment]
> > # Breaking Changes in MapboxNavigation
> > # Breaking API Changes
> > ## `NavigationMapView`
> > 
> > * _removed_ method: `showcase(_:animated:)` in `NavigationMapView`
> 
> At this point we will have to bump base version. Otherwise this will appear in every PR after this lands.
> 
> It is possible to not introduce such breaking change though. Adding a new method instead of modifying the old one. Even adding a new default parameter IS a breaking change for some code. An example:
> 
> ```swift
> import UIKit
> 
> // MARK: - v1 API
> 
> //func breakingMethod(param1: Bool) {
> //
> //}
> 
> // MARK: - v2 API
> 
> func breakingMethod(param1: Bool, param2: Bool = false) {
> 
> }
> 
> breakingMethod(param1: true)
> let methodReference = breakingMethod(param1:) // 1
> ```
> 
> `// 1` line works in v1 version of the API but doesn't work in v2.

@S2Ler, what do you mean by bumping base version? Changing to the most recent commit from `main` after landing this PR?

I think that this change to new default parameter should not bring any issues, as new parameter is completely different from what we had before. Adding new method for each change like this will not scale in long term, I guess.

File: Sources/MapboxNavigation/NavigationMapView.swift
Comment (1ec5) [reply Sources/MapboxNavigation/NavigationMapView.swift:131]: I’m sure this method is already used in the wild, but that doesn’t necessarily stop us from fixing what was essentially a bug in the method’s behavior (clipping the alternative routes). My inclination would be to frame it as a bug fix in the changelog, which would also skirt the questions around API stability below.

File: Sources/MapboxNavigation/NavigationMapView.swift
Comment (1ec5) [reply Sources/MapboxNavigation/NavigationMapView.swift:172]: This isn’t something I feel strongly about; I just pointed out the downsides for awareness. In my opinion, the fact that GitHub’s file viewing mode doesn’t wrap lines is a sufficient argument in favor of hard wrapping. I’m just less convinced about the other points, since generally, we can assume most developers will use Xcode’s default (which is to soft-wrap).

File: Sources/MapboxNavigation/NavigationMapView.swift
Comment (MaximAlien) [reply Sources/MapboxNavigation/NavigationMapView.swift]: Alright, updated method signature to use default argument to fit all routes to map. Changelog entry was updated as well.

File: Sources/MapboxNavigation/RoutesPresentationStyle.swift
Comment (S2Ler) [inline Sources/MapboxNavigation/RoutesPresentationStyle.swift:4]:
```suggestion
 A style that will be used when presenting routes on top of a map view by calling
```

File: Sources/MapboxNavigation/RoutesPresentationStyle.swift
Comment (S2Ler) [inline Sources/MapboxNavigation/RoutesPresentationStyle.swift:15]:
```suggestion
     All routes will be presented on a map view.
```

File: Sources/MapboxNavigation/NavigationMapView.swift
Comment (S2Ler) [reply Sources/MapboxNavigation/NavigationMapView.swift:174]: Even with wraps, the downside is that the text becomes hard to read because the text is too long[^1].

There are downsides as well as @1ec5 mentioned. But I think we should prioritize the decisions that make the lives of the devs who use our SDK easier/better. 

[^1]: [Readability of the text](https://ux.stackexchange.com/questions/108801/what-is-the-best-number-of-paragraph-width-for-readability)

**Review by avi-c — APPROVED** [review]
Other than the small syntax changes in the comments that Aliaksandr requested this looks good to me.

I'm approving pending their added approval.

**Comment by MaximAlien** [comment]
I think that all review comments were addressed, so I'm moving forward with merge. Feel free to comment if you'd like to add any improvements. I'll try to land them in subsequent updates.

**Pull request merged**
